homeassistant/components/homematic/cover.py

```diff
@@ -48,11 +54,20 @@ def set_cover_position(self, **kwargs):
 
     @property
     def is_closed(self):
-        """Return if the cover is closed."""
+        """Return whether the cover is closed."""
+        if self._hmdevice.__class__.__name__ in HM_GARAGE:
+            return self._hmdevice.is_closed(self._hm_get_state())
```

Comment:
Does this do I/O ? You are not allowed to do I/O in properties.

Reply:
Yes, like I said in my prev comments, you need use the values from data cache

Reply:
This call is not doing any I/O. It is just making the values from the state to a Boolean. It behaves like a static method, but did not want to make it really static in order to avoid including the pyhomematic class here.

I could just do the mapping here (IIRC 0 is closed, 1 and 2 is in-between and 3 is open; I don't remember exactly);.I just thought that the mapping belonged in the library and not here.